.github/workflows/build-wheel-macos-arm64.yaml

```diff
@@ -15,6 +15,8 @@ on:
 
 env:
   MACOSX_DEPLOYMENT_TARGET: 14.0
+  _PYTHON_HOST_PLATFORM: "macosx-14.0-arm64"
+  ARCHFLAGS: "-arch arm64"
```

Comment:
Do you think there is any way we could configure this in the build scripts (cmake, setup.py, etc), rather than in the build environment? Ideally without hardcoding the platform tag 🤔 

Right now this fix wouldn't cover local builds for example

Reply:
Hmm that's unfortunate. So we cannot generate the correct wheels locally without the user setting the env vars manually?

Also should we go for 14? I think we are targeting 13 in our releases https://pypi.org/project/PennyLane-Catalyst/#files

Reply:
Sure, we can try, but we should also lower the `MACOSX_DEPLOYMENT_TARGET: 14.0` to 13.0 then too.

Reply:
Assuming the 13.0 -> 14.0 bump here was needed, should we try this? https://github.com/PennyLaneAI/catalyst/commit/ece256518eda82847cd9474ad300663b668ff23a#diff-5ae2b3f6b392a8688e66c5d294dca714cae05c29f66952c1a22b7200625223c7R10

Reply:
Yeah the problem was that Rashid couldn't get the build respect the deployment target, since the device had macos 14 it would always build for that (don't know the details). 

The problem is if we name the wheels 14 then pip will flat out refuse to install them on a macos 13 system, but in my experience the 14 wheels have always worked on 13 (from when I was still running 13). In fact we have always had "14" wheels, but publish for 13 by [renaming them](https://github.com/PennyLaneAI/catalyst/blob/50781538350b659bdf5e708207daed8ba1bf4a15/.github/workflows/build-nightly-release.yaml#L76).

Reply:
So, just to clarify, we want to keep the MACOS target value at 14, but explicitly name the wheels as 13? And this hasn't caused any issues on Macs that we know of?

Reply:
I'm happy to lower it to 13, but as far as know it doesn't have any effect (according to Rashid). We definitely want to 13 in the wheel name though so that they are installable on macOS 13, never had any issues so far (we don't really use OS internals so that's probably why).

Reply:
I've coaxed it to set the value based on the (previously ignored) env var. Setuptools doesn't respect anything but the SDK version when number, which should get fixed by the final packaging step. I forced the CMake args to ensure `macosx-version-min=13.0` is set, and the wheel to take `_PYTHON_HOST_PLATFORM` which gets used during packaging. In essence, we are replicating a bunch of stuff usually handled by CIBuildWheel --- it may be worth relying on that for the final builds in future to help remove a bunch of this.

Reply:
Thank you!